### ISSUE TITLE
v2的MPI任务，设置sparse update后速度没提高

### ISSUE THREAD

**Issue by FrankRouter** [PaddlePaddle/Paddle#3175]
网络结构
输入为80万稀疏特征
一个隐层128维
softmax

损失函数
multiclass cross entropy

对比实验
实验零
v1版本，不设sparse update，一个pass约3小时
实验一
v1版本，设置sparse update，一个pass约半小时
实验二
v2版本，不设sparse update，一个pass约4.5小时
实验三
v2版本，设置sparse update，一个pass约4.5小时

问题
v2版本，训练速度没有提高

**Comment by jacquesqiao** [comment]
请问您是怎么设置的？

**Comment by PseudoProgrammer** [comment]
试验零和试验一是用paddle_1103版本提交的
1、试验零、trainer_config.conf

 91  Layer(inputs = [Input("input1", parameter_name = "_layer1_1.w", decay_rate_l1 = 0.01)], name = "layer1_1", bias = Bias(parameter_name = "_layer1_1.bias"), active_type = "tanh", type = "fc", size = 128)

2、试验一、trainer_config.conf

 91  Layer(inputs = [Input("input1", parameter_name = "_layer1_1.w", sparse_remote_update = True, decay_rate_l1 = 0.01)], name = "layer1_1", bias = Bias(parameter_name = "_layer1_1.bias"), active_type = "tanh", type = "fc", size = 128)

3、试验二、trainer_config.conf
 59     h = paddle.layer.fc(                                                         
 60         input=data,                                                              
 61         size=h_size,                                                             
 62         act=paddle.activation.Tanh(),                                            
 63         param_attr=paddle.attr.Param(initial_std=1.0 / math.sqrt(h_size)))

4、试验三、trainer_config.conf

 59     h = paddle.layer.fc(                                                         
 60         input=data,                                                              
 61         size=h_size,                                                             
 62         act=paddle.activation.Tanh(),                                            
 63         param_attr=paddle.attr.Param(initial_std=1.0 / math.sqrt(h_size), sparse_update=True))

**Comment by typhoonzero** [comment]
看这里，实验1是在_layer1_1.w这个层加了sparse update，而实验2是在fc加的？fc层本身不够稀疏，这样两个实验的网络是有区别的，可否保持实验2的网络配置一致再测试下呢？

**Comment by PseudoProgrammer** [comment]
那试验三的网络配置应该怎么配置才能对layer1_1.w加sparse_update，而不是对fc呢

**Comment by typhoonzero** [comment]
试下把`input=data`的`data`这层改成sparse呢？

**Comment by PseudoProgrammer** [comment]
我知道你的意思，但是不知道怎么配置，这样？
data = paddle.layer.data("word",                                             
                              paddle.data_type.sparse_binary_vector(dict_dim), param_attr=paddle.attr.Param(initial_std=1.0 / math.sqrt(h_size), sparse_update=True)) 

 h = paddle.layer.fc(                                                         
         input=data,                                                              
         size=h_size,                                                             
         act=paddle.activation.Tanh(),                                            
         param_attr=paddle.attr.Param(initial_std=1.0 / math.sqrt(h_size)))

**Comment by PseudoProgrammer** [comment]
data层应该没有初始化param_attr=paddle.attr.Param这个参数吧

**Comment by typhoonzero** [comment]
是有的：参考https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/layers.py#L876

和其他层是一样的配置。

**Comment by lcy-seso** [comment]
我觉得实验三的配置方法是没问题的。如果确实没有加速效果，也有可能是v2 本身实现时候，sparse update 有问题。这个需要一点时间排查。

**Comment by typhoonzero** [comment]
@FrankRouter 这几个实验是本地执行的还是集群方式执行的呢？

**Comment by FrankRouter** [comment]
集群。任务链接已经失效。需要的话，可以复现。

**Comment by Yancey1989** [comment]
Hi @FrankRouter ，方便的话把可以把Job链接发到我的Hi上么（yanxu05)

**Comment by zjgtan** [comment]
求问这个问题解决了吗？应该怎么设置sparse update

**Comment by Yancey1989** [comment]
根据Hi群里提供的复现链接来看，sparse update是生效的，需要在训练日志中区分实际training和test的时间。
我先关闭这个issue了，如果有进一步的问题可以再打开它。
